### PR TITLE
fix empty filenames are not considered a field (#26)

### PR DESCRIPTION
**Pull request by cyantree**
This PR normalizes empty filenames to be handled as `undefined`. This allows sending multipart requests from the browser containing fields with content types which then will be handled as fields and not as files.

Linting fails. I have fixed most of the issues locally however there are some missing dependencies regarding the benchmark tests. Should I add the fixes I made to this PR?

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by cyantree** [comment]
I've marked it as draft because I haven't tested the new version with my usecase.

**Comment by kibertoad** [comment]
You can ignore linting warnings for benchmarks, they will not fail the build, we'll fix them later.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,3 +9,4 @@ Major changes since the last busboy release (0.31):
 * Error on non-number limit rather than ignoring (#7)
 * Dicer is now part of the busboy itself and not an external dependency (#14)
 * Tests were converted to Mocha (#11, #12, #22, #23)
+* Empty filenames of parts are handled as `undefined` (#26)
```
Comment (kibertoad) [inline CHANGELOG.md:12]: can you also add entry to MIGRATING.md, explaining the breaking change?

File: test/types-multipart.spec.js [hunk 2]
```diff
@@ -267,12 +285,18 @@ describe('types-multipart', () => {
                     'Content-Type: application/json',
                     '',
                     '{}',
+                    '------WebKitFormBoundaryTB2MiQ36fnSJlrhY',
+                    'Content-Disposition: form-data; name="cont2"; filename=""',
+                    'Content-Type: text/javascript',
+                    '',
+                    'console.log("hello world");',
```
Comment (kibertoad) [inline test/types-multipart.spec.js:292]: is this intentional?

**Comment by kibertoad** [comment]
Linting fixes unrelated to this PR are very welcome, but having them as a separate PR would be more convenient.

**Comment by Uzlopak** [comment]
Is this relevant to this PR?
https://github.com/simllll/busboy/commit/c01a909b4bd901d644f1b9c2c4c1efeb48e1827e

**Comment by kibertoad** [comment]
@Uzlopak I'd say so, and going with a config options (with `blob` by default) sounds like a good solution.

**Comment by kibertoad** [comment]
@cyantree Please let me know if you'll need any help with this!

**Comment by cyantree** [comment]
> Is this relevant to this PR? [simllll@c01a909](https://github.com/simllll/busboy/commit/c01a909b4bd901d644f1b9c2c4c1efeb48e1827e)

Yes, this is relevant. However I wouldn't recommend using `blob` as identifier as it can easily be used as a filename and therefore would lead to unexpected behaviour.

Regarding introducing a config option:
Yes, I can see this. But I wouldn't use `blob` as default there either because of the above reasons.
I think empty filenames should be sufficient and are usable with the browser api.
Another option would be using filenames that can't exist because of their structure. Maybe something like `field://`.

If you want to fix this quickly, then yes, it would be great if you could take over because atm I only can work at this when time allows it.

File: test/types-multipart.spec.js (hunk 2 above)
Comment (cyantree) [reply test/types-multipart.spec.js:292]: What do you mean exactly? This tests whether it's possible to use a non-json text field with empty filename. Or do you think this test isn't necessary?

File: CHANGELOG.md (hunk 1 above)
Comment (cyantree) [reply CHANGELOG.md:12]: Yes, I can explain the breaking change.

**Comment by Uzlopak** [comment]
I created a new PR with your work #52 . It also ensures that we are not breaking with original busboy.

**Comment by Uzlopak** [comment]
I created another PR #53  which i think handles the whole blob issue correctly.

**Comment by cyantree** [comment]
Obsolete as of #53.

**Pull request closed without merging**